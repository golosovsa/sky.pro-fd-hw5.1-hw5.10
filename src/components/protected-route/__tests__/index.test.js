import { BrowserRouter } from 'react-router-dom'
import ShallowRenderer from 'react-test-renderer/shallow'
import ProtectedRoute from '..'

describe('<ProtectedRoute />', () => {
  describe('Snapshot test', () => {
    it('should render successfully and match snapshot', () => {
      const renderer = new ShallowRenderer()
      const tree = renderer.render(
        <BrowserRouter>
          <ProtectedRoute isAllowed={true}>123</ProtectedRoute>
        </BrowserRouter>
      )

      expect(tree).toMatchSnapshot()
    })
  })
})
