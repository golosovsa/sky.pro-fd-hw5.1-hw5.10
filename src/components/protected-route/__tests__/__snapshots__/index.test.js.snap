// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProtectedRoute /> Snapshot test should render successfully and match snapshot 1`] = `
<Router
  location={
    Object {
      "hash": "",
      "key": "default",
      "pathname": "/",
      "search": "",
      "state": null,
    }
  }
  navigationType="POP"
  navigator={
    Object {
      "action": "POP",
      "createHref": [Function],
      "encodeLocation": [Function],
      "go": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "default",
        "pathname": "/",
        "search": "",
        "state": null,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <UNDEFINED
    isAllowed={true}
  >
    123
  </UNDEFINED>
</Router>
`;
